lib/datacite/validators/attributes_validator.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'json_schemer'
+require 'pathname'
+
+module Datacite
+  module Validators
+    # Perform validation against openapi definition for the DataCite API
+    class AttributesValidator
+      def self.validate(attributes)
+        json_attributes = JSON.parse(attributes.to_json)
+        schemer = JSONSchemer.schema(json_schema_path)
+        results = schemer.validate(json_attributes)
+        return true if results.first.nil?
+
+        raise ValidationError, results.map { |result| result.fetch('error') }.join('\n')
+      end
+
+      # This json_schema file borrowed from https://github.com/inveniosoftware/datacite/blob/master/datacite/schemas/datacite-v4.5.json
+      # with the addition of the `identifiers` block
+      def self.json_schema_path
+        Pathname.new(File.expand_path('../../../datacite-v4.6.json', __dir__))
```

Comment:
Can this use `Rails.root` instead of relative path?

Reply:
As is, no - since this is a gem without Rails included yet. I work on that if we feel like it's important, but as is, no.